pkg/registration/registration_client.go

```diff
@@ -228,11 +228,47 @@ func (rClient *K8sRegistrationClient) GetActionMergePolicy() []*proto.ActionMerg
 		ForCommodity(proto.CommodityDTO_VMEM).
 		ForCommodity(proto.CommodityDTO_VCPU_REQUEST).
 		ForCommodity(proto.CommodityDTO_VMEM_REQUEST).
-		DeDuplicateAndAggregateBy(actionMergeTarget).
-		DeDuplicateAndAggregateBy(actionMergeTarget2)
+		DeDuplicateAndAggregateBy(resizeActionMergeTarget).
+		DeDuplicateAndAggregateBy(resizeActionMergeTarget2)
+
+	//horizontal scale action
+	horizontalScaleActionMergeTarget := builder.NewActionDeDuplicateAndAggregationTargetBuilder().
+		DeDuplicatedBy(builder.NewActionAggregationTargetBuilder(proto.EntityDTO_WORKLOAD_CONTROLLER,
+			proto.ConnectedEntity_AGGREGATED_BY_CONNECTION))
+
+	containerPodDataDaemonSet := proto.EntityDTO_ContainerPodData{
+		ControllerData: &proto.EntityDTO_WorkloadControllerData{
+			ControllerType: &proto.EntityDTO_WorkloadControllerData_DaemonSetData{
+				DaemonSetData: &proto.EntityDTO_DaemonSetData{
+					// Populate the fields specific to DaemonSetData
+				},
+			},
+		},
+	}
```

Comment:
Why do we specify daemonset here? There won't be SLO horizontal scale actions for daemonset.

Reply:
This is specified for the filter to not have the merged action: https://vmturbo.atlassian.net/wiki/spaces/AE/pages/3668901989/SLO+Horizontal+Scale+Action+Merging#Replicas-Represented-as-Attribute-on-WorkloadController
DaemonSets and ReplicaSets
Actions related to DaemonSets and ReplicaSets will not be merged. We should ignore any Provision/Suspend actions related to them during the merge process.